storage/maria/ha_maria.cc

```diff
@@ -3220,6 +3222,21 @@ void ha_maria::update_create_info(HA_CREATE_INFO *create_info)
   {
     create_info->auto_increment_value= stats.auto_increment_value;
   }
+  else if (thd_sql_command(ha_thd()) == SQLCOM_ALTER_TABLE)
```

Comment:
handler::update_create_info is also called from show_create_table, so you'll end up in a warning for every SHOW CREATE TABLE